src/client/Microsoft.Identity.Client/Http/HttpManager.cs

```diff
@@ -187,7 +193,7 @@ private HttpRequestMessage CreateRequestMessage(Uri endpoint, IDictionary<string
             }
 
             // package 500 errors in a "service not available" exception
-            if (isRetryableStatusCode)
+            if (isRetryableStatusCode && !isManagedIdentity)
```

Comment:
Because the builds exception from response which is OAuth specific. Adding this condition allows the exception to be handled for specific managed identity.